### ISSUE TITLE
Typing - Update weird behaviour with $in 

### ISSUE THREAD

**Issue by billouboq** [Automattic/mongoose#12049]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.4.2

### Node.js version

12.13.1

### MongoDB server version

4.4

### Description

Hi, I just used mongoose types and found some weird behaviour with the $in operator.

Here is the error :

```
error TS2322: Type '{ $in: mongoose.Types.ObjectId[]; }' is not assignable to type 'Partial<ObjectId> | { _id?: FilterOperators<ObjectId>; _bsontype?: FilterOperators<"ObjectID">; id?: FilterOperators<Buffer>; ... 6 more ...; inspect?: FilterOperators<...>; }'.
  Object literal may only specify known properties, and '$in' does not exist in type 'Partial<ObjectId> | { _id?: FilterOperators<ObjectId>; _bsontype?: FilterOperators<"ObjectID">; id?: FilterOperators<Buffer>; ... 6 more ...; inspect?: FilterOperators<...>; }'.

448   { $pull: { _directAccessUserGroupIds: { $in: groupIds } } },
```

The code is a simple :

```ts
User.findOneAndUpdate(
	{ _id: userId },
	{ $addToSet: { _directAccessUserGroupIds: groupIds } },
)
```

with groupIds being mongoose.Types.ObjectId[] type.

This seems to trigger everytime $in is used in an other operator.

### Steps to Reproduce

Use findOneAndUpdate with an "$in" inside an other operator

**Comment by billouboq** [comment]
Also I saw that VSCode is giving me ObjectId properties as suggestion so I don't know if it it takes ObjectId as type in those cases.

See this video 
https://user-images.githubusercontent.com/7765196/177114358-9d266839-5780-47a2-9ca2-05e4427c02d3.mov

**Comment by wight554** [comment]
Got similar issue with $pull in one-to-many relation
```
server/comment/CommentService.ts:117:9 - error TS2345: Argument of type '{ $pull: { comments: string; }; }' is not assignable to parameter of type 'UpdateWithAggregationPipeline | UpdateQuery<PostDocument> | undefined'.
  Types of property '$pull' are incompatible.
    Type '{ comments: string; }' is not assignable to type 'PullOperator<_UpdateQueryDef<PostDocument>>'.
      Type '{ comments: string; }' is not assignable to type '{ readonly _id?: FilterOperators<any> | Partial<any> | { [x: string]: FilterOperators<any> | undefined; } | undefined; readonly __v?: FilterOperators<any> | Partial<...> | { ...; } | undefined; readonly id?: FilterOperators<...> | ... 2 more ... | undefined; readonly comments?: FilterOperators<...> | ... 1 more ... ...'.
        Types of property 'comments' are incompatible.
          Type 'string' is not assignable to type 'FilterOperators<undefined> | Partial<Comment | undefined> | { text?: FilterOperators<string> | undefined; author?: FilterOperators<...> | undefined; ... 4 more ...; updatedAt?: FilterOperators<...> | undefined; }'.

117         {
            ~
118           $pull: { comments: commentId },
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
119         },
    ~~~~~~~~~
```

Code:
```
  PostModel.updateOne(
    { _id: postId },
    {
      $pull: { comments: commentId },
    },
    { useFindAndModify: false },
  );
```

**Comment by vkarpov15** [comment]
@billouboq it looks like this was fixed with #12042. I can confirm the below script fails to compile before #12042, but compiles fine after:

```ts
import mongoose from 'mongoose';

run().catch(err => console.log(err));

async function run() {
  const schema = new mongoose.Schema({ groupIds: ['ObjectId'] });
  const Test = mongoose.model('Test', schema);

  const newIds: mongoose.Types.ObjectId[] = [new mongoose.Types.ObjectId()];
  await Test.findOneAndUpdate({}, { $addToSet: { groupIds: newIds } });
}
```

**Comment by billouboq** [comment]
Wonderful !
Thank you very much !
